scripts/tests/test_zkai_attention_kv_model_faithful_quantized_attention_bridge_gate.py

```
@@ -0,0 +1,128 @@
+                gate.validate_result(result, payload, receipt)
+
+    def test_rejects_source_fixture_semantic_drift(self):
+        payload = copy.deepcopy(self.payload)
+        payload["score_rows"][0]["weight_denominator"] = 0
+        with self.assertRaisesRegex(gate.ModelFaithfulQuantizedAttentionBridgeGateError, "denominator|score rows|validation"):
+            gate.validate_result(self.result, payload, self.receipt)
+
+    def test_rejects_receipt_kernel_overclaim(self):
+        receipt = copy.deepcopy(self.receipt)
+        receipt["kernel_contract"]["real_softmax_status"] = "GO_REAL_VALUED_SOFTMAX"
+        with self.assertRaisesRegex(gate.ModelFaithfulQuantizedAttentionBridgeGateError, "kernel contract|receipt"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)